### ISSUE TITLE
Keras autoencoder bug?

### ISSUE THREAD

**Issue by kerasfun** [keras-team/keras#12394]
I tried the code snippet on the following webpage (https://blog.keras.io/building-autoencoders-in-keras.html) to build a simple autoencoder to detect outliers, but strangely the top three outliers have the lowest anomaly scores among the samples. It has only one hidden layer of 10 hidden neurons, and I used sigmoid as the activation functions. I tried Adam or SGD optimizers but it stayed the same.

Using the same neural network structure, I also tried the same thing in the MATLAB (https://www.mathworks.com/help/deeplearning/ref/trainautoencoder.html) and these three outliers are found to have the top three anomaly scores.

The Pearson correlation between the anomaly scores calculated by Keras or MATLAB is 0.34. While this correlation value indicates the trend of outlierness is shared by Keras and MATLAB results, but it is very strange that the Keras autoencoder models the ground truth outlier samples much better than regular normal samples, which is counterintuitive.
